typed/remove_test.go

```diff
@@ -0,0 +1,340 @@
+package typed
+
+import (
+	"fmt"
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/v3/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/v3/value"
+)
+
+func TestRemoveDeduced(t *testing.T) {
+
+	var cases = []struct {
+		object YAMLObject
+		remove *fieldpath.Set
+		expect YAMLObject
+	}{
+		{
+			object: `{}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			expect: `{}`,
+		},
+		{
+			object: `{"a": "value"}`,
+			remove: fieldpath.NewSet(fieldpath.MakePathOrDie("a")),
+			expect: ``,
```

Comment:
I'm really confused by the inconsistency here. If we have an empty object, we keep the empty object. But if we remove the single item from the object, we don't have an object anymore? :-))

Reply:
I think that's  something different, but let me verify.
As I remember it, it did not matter if expect is empty or {} because of the way this string gets parsed and compare works.
At least for deduced

Reply:
Ah, yeah, then I would make the tests consistent if possible